lib/mayaUsd/render/vp2RenderDelegate/draw_item.cpp

```diff
@@ -54,4 +51,20 @@ HdVP2DrawItem::~HdVP2DrawItem()
     }
 }
 
+void HdVP2DrawItem::AddRenderItem(MHWRender::MRenderItem* item, const HdGeomSubset* geomSubset)
+{
+    _renderItems.push_back(RenderItemData());
+    RenderItemData& renderItemData = _renderItems.back();
+
+    renderItemData._renderItem = item;
+    renderItemData._renderItemName = _drawItemName;
+    if (geomSubset) {
+        renderItemData._geomSubset = *geomSubset;
```

Comment:
Is there a reference-counting approach to hold the geomSubset object which can avoid the assignment?

Reply:
No there is no reference counting for [HdGeomSubset](https://github.com/PixarAnimationStudios/USD/blob/release/pxr/imaging/hd/geomSubset.h). The structure itself is made up of two SdfPaths and a VtIntArray, both of which have efficient operator equals, so the cost of making the copy is low.

Making the copy also substantially simplifies the code in the absence of ref counting, because I don't have to keep on going back to the topology for sub geom values.